tests/test_wave_05.py

```diff
@@ -80,34 +83,55 @@ def test_create_goal(client):
     }
 
 
-@pytest.mark.skip(reason="test to be completed by student")
+# @pytest.mark.skip(reason="test to be completed by student")
 def test_update_goal(client, one_goal):
-    raise Exception("Complete test")
+    # raise Exception("Complete test")
     # Act
     # ---- Complete Act Here ----
+    response = client.put("/goals/1", json={
+        "title": "Updated Goal Title"
+    })
+
+    response_body = response.get_json()
 
     # Assert
     # ---- Complete Assertions Here ----
     # assertion 1 goes here
+    assert response.status_code == 200
     # assertion 2 goes here
+    assert "goal" in response_body
     # assertion 3 goes here
+    assert response_body == {
+        "goal": {
+            "id": 1,
+            "title": "Updated Goal Title"
+        }
+    }
```

Comment:
So, as mentioned in your `Goal` model class, this test fails if you fill in the `"tasks"` key. Rather than _not_ returning this information, I would recommend updating this test.

A simple way would be to add the `"tasks"` key:

```python
assert response_body = {
    "goal": {
        "id": 1,
        "title": "Updated Goal Title",
        "tasks": []
     }
}
```

Which is OK in our case. But what would happen if we added a new key? And then another. It could be tiring to constantly update here, especially since the `"title"` key is the key we really care about for this test. So we could just check that directly with `assert response_body["goal"]["title"] == "Updated Goal Title"`, and then this test will pass no matter what additional keys we add.

However, there _is_ value in not just testing that the `"title"` updated but that the other keys did _not_ change. In that case, rather than comparing to a literal here, I would do something like this, where I compare it to the goal as it appears before updating:

```python
get_response = client.get("/goals/1/")
EXPECTED_TITLE = "Updated Goal Title"
expected_response = get_response.get_json()
expected_response["title"] = EXPECTED_TITLE

put_response = client.put("/goals/1/", json={
    "title": EXPECTED_TITLE
})
put_response_body = put_response.get_json()

assert put_response.status_code == 200
assert "goal" in put_response_body
assert put_response_body == expected response

goal = Goal.query.get(1)
assert goal.title == EXPECTED_TITLE
```

But there's an even better way! You'd have to change the `one_goal` fixture in conftest.py to return the goal that got created:

```python
@pytest.fixture
def one_goal(app):
    new_goal = Goal(title="Build a habit of going outside daily")
    db.session.add(new_goal)
    db.session.commit()
    return new_goal
```

Then , that `new_goal` will be passed in to your test as `one_goal`. Which you can then use instead of having to make a call to `GET /goals/1`:

```python
EXPECTED_TITLE = "Updated Goal Title"
expected_response = one_goal.to_dict()
expected_response["title"] = EXPECTED_TITLE

response = client.put("/goals/1/", json={
    "title": EXPECTED_TITLE
})
response_body = put_response.get_json()

assert response.status_code == 200
assert "goal" in response_body
assert response_body == expected response

goal = Goal.query.get(1)
assert goal.title == EXPECTED_TITLE
```

Of course _we instructors_ (well, not me personally) wrote the conftest.py file so this isn't really on you.